bus/broadcast_consumer_connection.go

```diff
@@ -52,6 +53,26 @@ func (b *Bus) doUpdateConnection(_ context.Context, msg *Message) error {
 }
 
 func (b *Bus) doDeleteConnection(ctx context.Context, msg *Message) error {
+	// Ensure this connection isn't being used by any dynamic replays
+	b.config.PersistentConfig.DynamicReplaysMutex.RLock()
+	for id, dynamicReplay := range b.config.PersistentConfig.Dynamic {
+		if dynamicReplay.Options.ConnectionId == id {
+			return fmt.Errorf("cannot delete connection '%s' because it is in use by dynamic replay '%s'",
+				id, dynamicReplay.Options.XDynamicId)
+		}
+	}
+	b.config.PersistentConfig.DynamicReplaysMutex.RUnlock()
+
+	// Ensure this connection isn't being used by any relays
+	b.config.PersistentConfig.RelaysMutex.RLock()
+	for id, dynamicReplay := range b.config.PersistentConfig.Relays {
+		if dynamicReplay.Options.ConnectionId == id {
+			return fmt.Errorf("cannot delete connection '%s' because it is in use by relay '%s'",
```

Comment:
Need to unlock here

Reply:
I would honestly just do a lock at the beginning (or a lock + copy) and defer unlock.